controller/lib/hal/buzzer.cpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#if defined(BARE_STM32)
+
+// The buzzer we use for generating alarms on the controller board is
+// part number CT-1205H-SMT-TR made by CUI.
+//
+// This part is designed to be driven with a 2.4kHz square wave.
+//
```

Comment:
Excuse my ignorance here. By reading the datasheet, I noticed that it said it was power rated under 2.4kHz and 0.5 duty cycle, but does that mean it cannot work at other frequencies? I also see there is a frequency response graph and there it it seems that has a pretty broad range at similar dB.

Reply:
I only tried it at 2.4kHz, so no doubt it could be used at other frequencies just with lower output.  Buzzers like this are designed to operate at a specific frequency, so normally that's how they're used.  I think we would be hard pressed to get much range out of it, it's not like a speaker that's designed to be used over a broad frequency range.

I'd prefer to keep this interface simple and not try to play music on the buzzer or something like that.  What did you have in mind?

Reply:
No, nothing so awesome in mind. But I do believe the ISO standards require specific tones for alarms. We might actually have to do that. Just wanted to make sure I understood the datasheet correctly, because the way you worded it seemed stronger than I read into it.